src/custom-layers/RadialGradientLayer/RadialGradientLayer.ts

```diff
@@ -174,6 +181,13 @@ export class RadialGradientLayer implements CustomLayerInterface {
    * @returns {Promise<void>} A promise that resolves when the animation completes
    */
   private async animateIn() {
+    if (!this.animationActive) {
+      this.scale = this.gradient.scale;
+      this.animationDelta = 1;
+      this.map.triggerRepaint();
+      return Promise.resolve();
```

Comment:
Is the delay introduced here (in comparison to just `return;`) needed for the logic? Same for `animateOut`. If yes, I think it warrants an explaining comment.

Reply:
Initially I thought returning a `Promise.resolve()` this would be more "intentful", but given there would be no functional difference between `return` nor `return Promise.resolve()` (the function is already async 🤷), it probably doesn't make sense to return the `Promise.resolve()` and just adds needless code.

Reply:
the return Promise.resolve() in animate out was to appease ts, but I'll refactor it to use `async` instead